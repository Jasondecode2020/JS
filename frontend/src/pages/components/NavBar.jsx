import React from 'react'
import { Nav, Navbar, Container } from 'react-bootstrap';
import Cookies from 'universal-cookie';
import { ReactComponent as Logo } from '../../logo.svg';

const NavBar = () => {
  const cookies = new Cookies();
  const token = cookies.get('user-token');

  /** clear token cookie to log out */
  function logOut () {
    cookies.remove('user-token');
    const logoutUrl = 'http://localhost:5005/admin/auth/logout';

    fetch(logoutUrl, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: token
      }
    }).then((res) => {
      if (res.status === 200) {
        alert('logout success!');
      } else {
        alert('ERROR: Invalid Input!');
      }
    }).catch((error) => {
      console.log(error);
    });
  }

  return (
    <Navbar collapseOnSelect expand="lg" bg="dark" variant="dark">
      <Container>
        <Navbar.Brand href="/dashboard"><Logo className="logo" />BigBrain</Navbar.Brand>
        <Navbar.Toggle aria-controls="responsive-navbar-nav" />
        <Navbar.Collapse id="responsive-navbar-nav">
          <Nav className="mr-auto">
            <Nav.Link href="/dashboard">Dashboard</Nav.Link>
            <Nav.Link href="/login" onClick={ logOut }>Log out</Nav.Link>
          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  )
}

export default NavBar
