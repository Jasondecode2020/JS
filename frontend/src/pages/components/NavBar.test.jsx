import React from 'react';
import { shallow } from 'enzyme';
import NavBar from './NavBar';
import { Nav } from 'react-bootstrap';
import renderer from 'react-test-renderer';

describe('NavBar', () => {
  const components = shallow(<NavBar />);
  const wrapper = components.find('Logo');
  it('include the length of input is 0', () => {
    expect(wrapper.length).toBe(0);
  });

  it('include the text, BigBrain', () => {
    const NavText = 'BigBrainTo be addedLog out';
    expect(components.text()).toBe(NavText);
  });

  // test button click handler using enzyme
  it('include the button click event', () => {
    const onClick = jest.fn();
    shallow(<button onClick={onClick} />).simulate('click');
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  // test link
  it('include the Nav.Link text', () => {
    const NavLinkText = 'To be added';
    const newWrapper = shallow(<Nav.Link href='#features'>To be added</Nav.Link>)
    expect(newWrapper.text()).toBe(NavLinkText);
  });

  // snapshot
  it('renders with minimal props', () => {
    const logout = () => {
      console.log('check logout');
    }
    const button = renderer.create(<Nav.Link href='/login' onClick={ logout }>Log out</Nav.Link>).toJSON();
    expect(button).toMatchSnapshot();
  })
});
