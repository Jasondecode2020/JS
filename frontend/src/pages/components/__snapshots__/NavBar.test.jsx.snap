// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders with minimal props 1`] = `
<a
  className="nav-link"
  href="/login"
  onClick={[Function]}
  onKeyDown={[Function]}
>
  Log out
</a>
`;
